src/backend/seq/genpup.py

```diff
@@ -340,7 +341,21 @@ def switcher(typelist, pupstr, nests):
 ##### main function
 ########################################################################################
 if __name__ == '__main__':
-    #### generate the core pack/unpack kernels
+    ##### parse user arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--pup-max-nesting', type=int, help='maximum nesting levels to generate', required=True)
+    args = parser.parse_args()
+    if (args.pup_max_nesting < 0):
+        parser.print_help()
+        print
+        print("===> ERROR: pup-max-nesting must be positive")
+        sys.exit(1)
+
+    ##### generate the list of derived datatype arrays
+    darraylist = [ ]
+    yutils.generate_darrays(derived_types, darraylist, args.pup_max_nesting)
+
+    ##### generate the core pack/unpack kernels
```

Comment:
Is it easy to factor out a common module and avoid duplication?

Reply:
`yutils` is the common module.  It's meant for functions that are generic across all `yaksa` scripts.